quinn-proto/src/shared.rs

```diff
@@ -183,24 +183,35 @@ pub enum ConfigError {
 
 /// Events to be sent to the Connection
 pub enum ConnectionEvent {
+    /// A datagram has been received for the Connection
     Datagram {
+        #[doc(hidden)]
         now: Instant,
+        #[doc(hidden)]
         remote: SocketAddr,
+        #[doc(hidden)]
         ecn: Option<EcnCodepoint>,
+        #[doc(hidden)]
         first_decode: PartialDecode,
+        #[doc(hidden)]
         remaining: Option<BytesMut>,
     },
+    /// New connection identifiers have been issued for the Connection
     NewIdentifiers(Vec<(u64, ConnectionId)>),
+    /// A timeout has fired for a Connection
     Timer(Instant, Timer),
 }
 
 /// Events to be sent to the Endpoint
 #[derive(Clone, Debug)]
 pub enum EndpointEvent {
```

Comment:
I feel like we could reduce the API surface area by wrapping this in an opaque newtype (and relying on `Connection::is_drained` instead of matching on events in high level code)

Reply:
Do you think that would make the API easier to understand? I actually feel like just looking at the different event enums and how they flow between tasks will make it straightforward to explain what's going on here. Maybe we should just `#[doc(hidden)]` the variants here?

Reply:
All of these strike me as fairly obscure details of QUIC from a user's perspective; I'm not sure it's going to mean much to someone who's not familiar with the drained state, stateless resets, or the CID lifecycle.